pcf/particle/aws/glacier/glacier_vault.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from botocore.exceptions import ClientError
+from pcf.core.aws_resource import AWSResource
+from pcf.core import State
+from pcf.util import pcf_util
+
+logger = logging.getLogger(__name__)
+
+class GlacierVault(AWSResource):
+    """
+    This is the implementation of Amazon's Glacier service. Additional functions that are callable on this particle
+    are upload_archive, initiate_job, list_jobs, delete_archive, delete_job, add_tags_to_vault, list_tags_for_vault
+    """
+    flavor = "glacier"
+
+    state_lookup = {
+        "missing": State.terminated,
+        "active": State.running,
+        "inactive": State.terminated
+    }
+
+    equivalent_states = {
+        State.running: 1,
+        State.stopped: 0,
+        State.terminated: 0
+    }
+
+    START_PARAMS_FILTER = {
+        "vaultName" # Required
+    }
+
+    def __init__(self, particle_definition):
+        super(GlacierVault, self).__init__(
+            particle_definition=particle_definition,
+            resource_name="glacier",
+        )
+        self.vault_name = self.desired_state_definition["vaultName"]
+
+    def _terminate(self):
+        """
+        Deletes the Glacier vault if vault is empty
+
+        Returns:
+             response of boto3 delete_vault
+        """
+        return self.client.delete_vault(vaultName=self.vault_name)
+
+    def _start(self):
+        """
+        Creates the Glacier vault
+
+        Returns:
+             response of boto3 create_vault
+        """
+        create_definition = pcf_util.param_filter(self.get_desired_state_definition(), self.START_PARAMS_FILTER)
+        response = self.client.create_vault(**create_definition)
+
+        if self.custom_config.get("Tags"):
+            tags = self.custom_config.get("Tags")
+
+            self.client.add_tags_to_vault(
+                vaultName=self.vault_name,
+                Tags=tags
+            )
+        return response
+
+
+    def _stop(self):
+        """
+        Glacier does not have a stopped state so it calls terminate.
+        """
+        return self.terminate()
+
+    def get_status(self):
+        """
+        Determines if the vault exists
+
+        Returns:
+             status (dict)
+        """
+
+        try:
+            vault_object = self.client.describe_vault(
+                vaultName=self.vault_name,
+                accountId="-"
```

Comment:
should take in account id from confi and default to `-`. helps if they add more to api going forward

Reply:
This is a GREAT point, thanks for catching that!